server/pulp/server/managers/repo/unit_association_query.py

```diff
@@ -349,18 +350,26 @@ def _associated_units_by_type_cursor(unit_type_id, criteria, associated_unit_ids
         :type associated_unit_ids: list
         :rtype: pymongo.cursor.Cursor
         """
-
+        model = get_unit_model_by_id(unit_type_id)
         collection = types_db.type_units_collection(unit_type_id)
```

Comment:
Consider:
```
if hasattr(model, 'SERIALIZER'):
    serializer = model.SERIALIZER()
else:
    serializer = None
```
and replace `if hasattr()` code below with `if serializer is not None` and call to `serializer.translate_filters()`.  This way the `hasattr()` check and serializer instantiation only happens once.